### PR TITLE
Remove all mentions  of 'Metal' by name

### PR DESCRIPTION
**Pull request by litherum**
***
<a href="https://pr-preview.s3.amazonaws.com/litherum/gpuweb/pull/2028.html" title="Last updated on Aug 11, 2021, 8:36 PM UTC (877cd8c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/2028/ef3fbc5...litherum:877cd8c.html" title="Last updated on Aug 11, 2021, 8:36 PM UTC (877cd8c)">Diff</a>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -3503,10 +3500,6 @@ truly optional.
         Indicates the required {{GPUTextureViewDescriptor/dimension}} for texture views bound to
         this binding.
 
-        Note:
-            This enables Metal-based WebGPU implementations to back the respective bind groups with
-            `MTLArgumentBuffer` objects that are more efficient to bind at run-time.
-
```
Comment (kainino0x) [inline spec/index.bs:3509]: Please leave a link back to https://github.com/gpuweb/gpuweb/pull/339 in a `<!-- comment -->` or `Note:`

File: spec/index.bs [hunk 2]
```diff
@@ -3543,10 +3536,6 @@ truly optional.
     ::
         Indicates the required {{GPUTextureViewDescriptor/dimension}} for texture views bound to
         this binding.
-
-        Note:
-            This enables Metal-based WebGPU implementations to back the respective bind groups with
-            `MTLArgumentBuffer` objects that are more efficient to bind at run-time.
```
Comment (kainino0x) [inline spec/index.bs:3549]: same

**Review by kainino0x — COMMENTED** [review]
I don't know the motivation for this change, but it seems acceptable. Do we need to avoid adding references to Metal even in `<!-- comments -->` in the future?

**Comment by kvark** [comment]
I'd love us to approach this consistently, i.e. by having rules applied to all of the APIs we target. Something like "mentioning native APIs can only be done in non-normative sections/comments".

**Comment by kvark** [comment]
Also, it would be helpful to know "why" this change is made, at least on some level.

**Comment by litherum** [comment]
> I don't know the motivation for this change, but it seems acceptable. Do we need to avoid adding references to Metal even in `<!-- comments -->` in the future?

Ideally, yes.

> I'd love us to approach this consistently, i.e. by having rules applied to all of the APIs we target. Something like "mentioning native APIs can only be done in non-normative sections/comments".

We have no opinion about mentioning other APIs; our request is just about Metal specifically. If the CG feels that other APIs should follow suit, that's fine, but orthogonal to our desire here.

> Also, it would be helpful to know "why" this change is made, at least on some level.

Please attend office hours tomorrow.

**Review by kainino0x — APPROVED** [review]

**Pull request merged**